src/napari/layers/shapes/_shapes_utils.py

```diff
@@ -253,62 +255,93 @@ def lines_intersect(p1, q1, p2, q2):
     intersects : bool
         Bool indicating if line segment p1q1 intersects line segment p2q2
     """
+    # Wrap inputs into (1, 2) arrays to use the vectorized function
+    p1_arr = np.array([p1])
+    q1_arr = np.array([q1])
+
+    return vectorized_lines_intersect(p1_arr, q1_arr, p2, q2)[0]
+
+
+def vectorized_lines_intersect(
+    p1: npt.NDArray, q1: npt.NDArray, p2: npt.NDArray, q2: npt.NDArray
+) -> npt.NDArray:
+    """Determines if a set of N line segments intersects with a single line segment.
+
+    This implementation is vectorized to be fast for many segments. It uses
+    the 'orientation' method, which is a vectorized version of the algorithm
+    used elsewhere in this codebase, making it easier to review and maintain.
+
+    The algorithm is based on checking if the endpoints of each segment lie on
+    opposite sides of the other segment.
+
+    Parameters
+    ----------
+    p1 : (N, 2) array
+        Array of first points of N line segments.
+    q1 : (N, 2) array
+        Array of second points of N line segments.
+    p2 : (2,) array
+        First point of the single line segment.
+    q2 : (2,) array
+        Second point of the single line segment.
+
+    Returns
+    -------
+    intersects : (N,) array of bool
+        Array with `True` values for line segments in the set that intersect
+        the single line segment.
+    """
     # Determine four orientations
-    o1 = _triangulate_py.orientation(p1, q1, p2)
-    o2 = _triangulate_py.orientation(p1, q1, q2)
-    o3 = _triangulate_py.orientation(p2, q2, p1)
-    o4 = _triangulate_py.orientation(p2, q2, q1)
+    o1 = _triangulate_py.vectorized_orientation(p1, q1, p2)
+    o2 = _triangulate_py.vectorized_orientation(p1, q1, q2)
+    o3 = _triangulate_py.vectorized_orientation(p2, q2, p1)
+    o4 = _triangulate_py.vectorized_orientation(p2, q2, q1)
 
     # Test general case
-    if (o1 != o2) and (o3 != o4):
-        return True
+    intersects = (o1 != o2) & (o3 != o4)
 
-    # Test special cases
+    # Test spacial cases
     # p1, q1 and p2 are collinear and p2 lies on segment p1q1
-    if o1 == 0 and on_segment(p1, p2, q1):
-        return True
+    intersects |= (o1 == 0) & on_segment(p1, p2, q1)
 
     # p1, q1 and q2 are collinear and q2 lies on segment p1q1
-    if o2 == 0 and on_segment(p1, q2, q1):
-        return True
+    intersects |= (o2 == 0) & on_segment(p1, q2, q1)
 
     # p2, q2 and p1 are collinear and p1 lies on segment p2q2
-    if o3 == 0 and on_segment(p2, p1, q2):
-        return True
+    intersects |= (o3 == 0) & on_segment(p2, p1, q2)
 
     # p2, q2 and q1 are collinear and q1 lies on segment p2q2
-    if o4 == 0 and on_segment(p2, q1, q2):  # noqa: SIM103
-        return True
+    intersects |= (o4 == 0) & on_segment(p2, q1, q2)
+
+    return intersects
 
-    # Doesn't fall into any special cases
-    return False
 
+def on_segment(p: npt.NDArray, q: npt.NDArray, r: npt.NDArray) -> npt.NDArray:
```

Comment:
This helper wasn't used anywhere else, so I felt it was Ok to just go with the refactor.